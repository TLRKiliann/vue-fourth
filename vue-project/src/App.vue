<script lang="ts">
import FirstComp from './components/FirstComp.vue'
import PopupComp from './components/PopupComp.vue'
import Card from './components/Card.vue';
import { defineComponent } from 'vue';
export default defineComponent({
  name: 'App',
  components: {
    FirstComp,
    PopupComp,
    Card,
},
  data() {
    return {
      showPopup: false,
    }
  },
  provide: {
    username: 'Esteban'
  }
})
</script>

<template>
  <FirstComp id='my-article' title="Esteban in Article" :likes="50" :isPublished="true" />

  <div>
    <button class="btn-one" @click="showPopup = true">Shwo my popup</button>
    <button class="btn-two" @click="showPopup = false">Hide my popup</button>
    <PopupComp v-show="showPopup" />
  </div>

  <div>
    <Card></Card>
    <Card>Header</Card>
    <Card>Main</Card>
    <Card>Footer</Card>
  </div>

  <Card>
    <template v-slot:header>
      <h2>Card Header</h2>
    </template>

    <template v-slot:main>
      <h2>Card Main</h2>
    </template>

    <template v-slot:footer>
      <h2>Card Footer</h2>
    </template>
  </Card>

</template>

<style scoped>
</style>
