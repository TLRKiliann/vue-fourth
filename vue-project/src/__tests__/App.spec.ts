import { describe, it, expect, vi } from 'vitest'

import { mount } from '@vue/test-utils'
import App from '../App.vue'

describe('App', () => {
  it('snapshot UI testing', () => {
    const wrapper = mount(App, {});
    expect(wrapper.text()).toMatchSnapshot()
  })
  it('renders properly', async () => {
    const wrapper = mount(App)
    await wrapper.find('.btn-one').trigger('click')
  })
  it('renders properly', async () => {
    const wrapper = mount(App)
    await wrapper.find('.btn-two').trigger('click')
  })
  it('should render Foo, then hide it', async () => {
    const wrapper = mount(App)
    const addOne = false;
    wrapper.vm.addOne = vi.fn();
    expect(wrapper.vm.addOne).toHaveBeenCalledTimes(0);
  })
})
