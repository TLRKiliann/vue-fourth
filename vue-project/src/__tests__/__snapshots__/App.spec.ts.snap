// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > snapshot UI testing 1`] = `"Hello ! Esteban in ArticleLikes: 50Published: yesSecondCompThirdComp EstebanShwo my popupHide my popupMy popupDefault Slot CardHeaderMainFooterDefault Slot CardCard HeaderCard MainCard Footer"`;
